string[] obj
string[] distance
string[] direction
int16[] Ximg
int16[] Yimg
